pkg/gitnessclient/gitness_client.go

```diff
@@ -0,0 +1,413 @@
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	var apiMemberships []apiMembershipResponse
+	if err := json.Unmarshal(body, &apiMemberships); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal response body: %w", err)
+	}
+
+	return apiMemberships, nil
+}
+
+func (g *GitnessClient) GetUser() (*apiUserResponse, error) {
+	userURL, err := g.BaseURL.Parse("/api/v1/user")
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse url : %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(context.Background(), "GET", userURL.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	req.Header.Set("accept", "application/json")
+	req.Header.Set("Authorization", "Bearer "+g.Token)
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to perform request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+	var apiUser apiUserResponse
+	if err := json.Unmarshal(body, &apiUser); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal response body: %w", err)
+	}
+
+	return &apiUser, nil
+}
+
+func (g *GitnessClient) GetRepositories(namespace string) ([]ApiRepository, error) {
+	space := ""
+	if namespace == personalNamespaceId {
+		user, err := g.GetUser()
+		if err != nil {
+			return nil, err
+		}
+		space = user.UID
+	} else {
+		space = namespace
+	}
+
+	reposURL, err := g.BaseURL.Parse(fmt.Sprintf("/api/v1/spaces/%s/+/repos", space))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse url : %w", err)
+	}
+
+	req, err := http.NewRequest("GET", reposURL.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("accept", "application/json")
+	req.Header.Set("Authorization", "Bearer "+g.Token)
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failed to fetch repositories, status code: %d", resp.StatusCode)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var apiRepos []ApiRepository
+	if err := json.Unmarshal(body, &apiRepos); err != nil {
+		return nil, err
+	}
+
+	return apiRepos, nil
+}
+
+func (g *GitnessClient) GetRepoBranches(repositoryId string, namespaceId string) ([]*apiRepoBranch, error) {
+	branchesURL, err := g.BaseURL.Parse(fmt.Sprintf("/api/v1/repos/%s/branches", url.PathEscape(namespaceId+"/"+repositoryId)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse url : %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(context.Background(), "GET", branchesURL.String(), nil)
```

Comment:
Since every method of the client preforms an http request in the same form, I suggest you create a utility method that will preform a request to the desired URL and return the body.

E.g.
```golang
respBody, err := g.preformRequest(method, URL)
....
```

This will make the client much more clean.

Reply:
@Tpuljak  Updating in miniutes

Reply:
@Tpuljak  Updated the code we can merge now.